### PR TITLE
Added: SQLite Database Support to LoveStack Edition of eZ Publish

### PR DESCRIPTION
**Pull request by se7enxweb**
Hello Mugo!

I write today with a big improvement to eZ Publish that I am excited to share.

With a little help from the community contribution corner I implemented SQLite Database Support for the eZ Publish Kernel and Setup Wizard. These changes are already merged into eZ Publish 6.0 GitHub and Netgen eZ Publish GitHub.

Please let me know what you think? We wanted to ensure that the top distributions of eZ Publish get the new database features. Thanking you in advance for your continued support!

Cheers,
7x

### CODE REVIEW

File: kernel/setup/steps/ezstep_database_choice.php [hunk 1]
```diff
@@ -78,7 +78,7 @@ function init()
             }
         }
 
-        if ( $databaseCount != 1 )
```
Comment (pkamps) [inline kernel/setup/steps/ezstep_database_choice.php:81]: Could you explain that change? Why was it necessary to invert this?

File: kernel/setup/ezsetupcommon.php [hunk 2]
```diff
@@ -29,7 +29,13 @@ function eZSetupOptionalTests()
 
 function eZSetupDatabaseMap()
 {
-    return array( 'mysqli' => array( 'type' => 'mysqli',
+    return array( 'sqlite' => array( 'type' => 'sqlite',
```
Comment (pkamps) [inline kernel/setup/ezsetupcommon.php:32]: Is that affecting the order in the frontend? Is 'sqlite' the first choice? I think we want to have it as the 3rd choice.

**Review by pkamps — COMMENTED** [review]
This looks very interesting - could you answer me those few questions? Thanks!

**Comment by se7enxweb** [comment]
@pkamps thank you for your questions.

I am away this week on birthday holiday with out of town family but I will address each of your concerns just as soon as possible.

Thanks again for your continued support.

Cheers
7x

**Comment by se7enxweb** [comment]
Hello @pkamps 

I am sorry for my delay. Your gentile challenge prompted me to do a lot more development and testing of the entire installation process (and options / paths). This time the work is much better tested for both mysql and sqlite installation paths to successful completion (a setup website).

The two outdated comments should be noted that I've addressed them by fixing the implementation and making the requested changes.

In short, please accept a much more tested and refined sqlite support with a focus on mysql as a first class citizen.

Please feel free to ask more questions! I'm studied up on the related code now. :)

Cheers,
7x

File: kernel/setup/ezsetupcommon.php (hunk 2 above)
Comment (se7enxweb) [reply kernel/setup/ezsetupcommon.php]: Addressed by moving sqlite down in order.

File: kernel/setup/steps/ezstep_database_choice.php (hunk 1 above)
Comment (se7enxweb) [reply kernel/setup/steps/ezstep_database_choice.php]: In testing in the short term it was required, but only because the implementation was not whole. I have reverted this change.

**Comment by se7enxweb** [comment]
I took some extra time and fixed a few more default installation errors that prevented non-developer users from completing a default mysql installation (with the sqlite changes in place) and with existing packages downloaded from the last installation use by the setup wizzard (after running the setup wizard a second time without removing the downloaded packages (as most users forget to do this step themselves).

I hope you can also see the value of these fixes to the default installation process.

Cheers,
7x

**Comment by se7enxweb** [comment]
Thank you so very much I appreciate you taking the time to review and merge this pr.

Happy Valentine's Day!

Cheers,
7x
